File: examples/python/find_events.py
from gi.repository import Zeitgeist, GLib
log = Zeitgeist.Log.get_default()
mainloop = GLib.MainLoop()

def on_events_received(log, result, data):
    events = log.find_events_finish(result)
    for i in xrange(events.size()):
        event = events.next_value()
        if event:
            print "Event id:", event.get_property("id")
            for i in xrange(event.num_subjects()):
                subj = event.get_subject(i)
                print "  -", subj.get_property("uri")
    mainloop.quit()

subject = Zeitgeist.Subject.full("", Zeitgeist.AUDIO, "", "", "", "", "")
event = Zeitgeist.Event()
event.add_subject(subject)
time_range = Zeitgeist.TimeRange.anytime ();

log.find_events(time_range, 
                        [event],
                        Zeitgeist.StorageState.ANY,
                        20,
                        Zeitgeist.ResultType.MOST_RECENT_SUBJECTS,
                        None,
                        on_events_received,
                        None)

mainloop.run()

File: examples/js/find_events.js
const Gtk = imports.gi.Gtk;
const Lang = imports.lang;
const Zeitgeist = imports.gi.Zeitgeist;

let zglog = Zeitgeist.Log.get_default();

let ids = [];
for (let i=200; i<222; i++){
    ids.push(i);
}

zglog.get_events(ids, null, Lang.bind(this,
            function(zg, result, data) {
                print ("===")
                let events = zg.get_events_finish(result);
                for (let i=0; i<events.size(); i++) {
                        print (i)
                        print(events[i]);
                 }
            }),
            null);

Gtk.main();
